h2o-core/src/main/java/hex/grid/filter/KeepOnlyFirstMatchFilterFunction.java

```diff
@@ -35,9 +35,10 @@ public boolean test(MP permutation) {
     return _baseMatchFunction.apply(permutation);
   }
 
+
   @Override
-  public void activate() {
-    if (_maxNumberOfMatchesToApply > 0) _maxNumberOfMatchesToApply--;
+  public void activate(boolean globalActivate, MP permutation) {
+    if (test(permutation) && globalActivate && _maxNumberOfMatchesToApply > 0) _maxNumberOfMatchesToApply--;
```

Comment:
can't you put `test(permutation)` last to avoid unnecessary computation?

Reply:
also, see comment above on the interface:
old `apply` could be the new `test`
old `test` can now be `private boolean internalTest()`
so that you expose only one method with `boolean xxx(MP)` signature

Reply:
@seb-h2o yes, I wanted to do this. Just pushed what I had before the call.